tests/catch/unit/memory/hipMalloc3DArray.cc

```diff
@@ -176,20 +126,144 @@ TEST_CASE("Unit_hipMalloc3DArray_MultiThread") {
     t.join();
   }
-  HIP_CHECK(hipMemGetInfo(&avail, &tot));
+  const auto avail = getFreeMem();
 
   if (pavail != avail) {
     WARN("Memory leak of hipMalloc3D API in multithreaded scenario");
     REQUIRE(false);
   }
 }
 
+void checkArrayIsExpected(hipArray_t array, const hipChannelFormatDesc& expected_desc,
+                          const hipExtent& expected_extent, const unsigned int expected_flags) {
+  std::ignore = array;
+  std::ignore = expected_desc;
+  std::ignore = expected_extent;
+  std::ignore = expected_flags;
+  // hipArrayGetInfo doesn't currently exist (EXSWCPHIPT-87)
+
+  // hipChannelFormatDesc queried_desc;
+  // hipExtent queried_extent;
+  // unsigned int queried_flags;
+
+  // hipArrayGetInfo(queried_desc, queried_extent, queried_flags, array);
+
+  // REQUIRE(expected_desc.x == queried_desc.x);
+  // REQUIRE(expected_desc.y == queried_desc.y);
+  // REQUIRE(expected_desc.z == queried_desc.z);
+  // REQUIRE(expected_desc.f == queried_desc.f);
+
+  // REQUIRE(expected_extent.width == queried_extent.width);
+  // REQUIRE(expected_extent.height == queried_extent.height);
+  // REQUIRE(expected_extent.depth == queried_extent.height);
+
+  // REQUIRE(expected_flags == queried_flags);
+}
+
+TEMPLATE_TEST_CASE("Unit_hipMalloc3DArray_happy", "", char, uchar2, uint2, int4, short4, float,
+                   float2, float4) {
+  hipArray_t array;
+  const auto desc = hipCreateChannelDesc<TestType>();
+#if HT_AMD
+  const unsigned int flags = hipArrayDefault;
+#else
+  const unsigned int flags = GENERATE(hipArrayDefault, hipArraySurfaceLoadStore);
+#endif
+  constexpr size_t size = 64;
+  hipExtent extent;
+
+  SECTION("1D Array") {
+    extent = make_hipExtent(size, 0, 0);
+    HIP_CHECK(hipMalloc3DArray(&array, &desc, extent, flags));
+  }
+  SECTION("2D Array") {
+    extent = make_hipExtent(size, size, 0);
+    HIP_CHECK(hipMalloc3DArray(&array, &desc, extent, flags));
+  }
+  SECTION("3D Array") {
+    extent = make_hipExtent(size, size, size);
+    HIP_CHECK(hipMalloc3DArray(&array, &desc, extent, flags));
+  }
+
+  checkArrayIsExpected(array, desc, extent, flags);
+
+  HIP_CHECK(hipFreeArray(array));
+}
+
+TEMPLATE_TEST_CASE("Unit_hipMalloc3DArray_MaxTexture", "", int, uint4, short, ushort2,
+                   unsigned char, float, float4) {
+#if HT_AMD
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-97");
+#endif
+
+  hipArray_t array;
+  const hipChannelFormatDesc desc = hipCreateChannelDesc<TestType>();
+#if HT_AMD
+  const unsigned int flag = hipArrayDefault;
+#else
+  const unsigned int flag = GENERATE(hipArrayDefault, hipArraySurfaceLoadStore);
+#endif
+  if (flag == hipArraySurfaceLoadStore) {
+    HipTest::HIP_SKIP_TEST("EXSWCPHIPT-58");
+    return;
+  }
+  CAPTURE(flag);
+  const Sizes sizes(flag);
+  CAPTURE(sizes.max1D, sizes.max2D, sizes.max3D);
+
+  const size_t s = 64;
+  SECTION("Happy") {
+    // stored in a vector so some values can be ifdef'd out
+    std::vector<hipExtent> extentsToTest{
```

Comment:
Suggestion: 
```cpp
      int x = GENERATE(s, sizes.max1D, sizes.max2D[0], sizes.max3D[0]);
      int y = GENERATE(0, s, sizes.max2D[1], sizes.max3D[1]);
      int z = GENERATE(0, s, sizes.max3D[2]);
      hipExtent extentToTest = make_hipExtent(x,y,z);
```

Reply:
this would lead to mixing of max1D, max2D and max3D e.g. `make_hipExtent(sizes.max1D, sizes.max2D[1], sizes.max3D[2])`